Kinvey.Core/Store/MultiInsertRequest.cs

```diff
@@ -125,13 +125,24 @@ public override async Task<KinveyMultiInsertResponse<T>> ExecuteAsync()
                     {
                         exception = httpRequestException;
                     }
+                    catch (KinveyException kinveyException)
```

Comment:
instead of having 2 `catch` statements, couldn't we just `catch (Exception)`?

Reply:
@heyzooi 
If we have HttpRequestException, we should swallow it. In other words a user will not know anything about occurring the exception. He will operate with  local data instead of network data.
If we have KinveyException, we need to throw it again since a user should know about it.
It can be 400(Bad request), 401(Invalid permissions),etc.

Reply:
@eugeninnowise What happens in the case where neither an `Exception` should be caught, but it is neither `HttpRequestException` or `KinveyException`?

Reply:
@vinaygahlawat 
A user will have an  unhandled exception.It means no data will be put to pending actions.
When I was developing this functionality, I though about this case.
The matter is if we have an Exception which is neither HttpRequestException nor KinveyException, it means something wrong in our implementation. In this case it would be better not to hide it by catching. In our Kinvey solution we don't have hi level try-catch{Exception}.
But basically I can suggest the solution where we catch an Exception. After this we will create an KinveyException based on it and throw the new one.
WDYT?

Reply:
@eugeninnowise I think in this case, it is better to leave your implementation as-is, where we handle `HttpResponseException` and `KinveyException` explicitly.